components/director/internal/domain/application/repository.go

```diff
@@ -57,12 +67,71 @@ func (r *inMemoryRepository) List(ctx context.Context, tenant string, filter []*
 	}, nil
 }
 
-// TODO: Make filtering and paging
-func (r *inMemoryRepository) ListByRuntimeID(ctx context.Context, tenant, runtimeID string, pageSize *int, cursor *string) (*model.ApplicationPage, error) {
+// TODO: @dbadura add pagination when PR-181 is merged
+func (r *inMemoryRepository) ListByScenariosForRuntime(ctx context.Context, tenantID string, runtimeID string, pageSize *int, cursor *string) (*model.ApplicationPage, error) {
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "while fetching DB from context")
+	}
+
+	//TODO: change tenantID from UUID to String when
+	tenantUUID, err := uuid.Parse(tenantID)
+	if err != nil {
+		return nil, errors.New("tenant_ID is not parseable")
+	}
+
+	stmt := fmt.Sprintf(`SELECT VALUE FROM %s WHERE TENANT_ID=$1 AND RUNTIME_ID=$2 AND KEY='SCENARIOS'`, labelTableName)
```

Comment:
it is really strange that your application repository is responsible for searching in labels table. Getting scenarios values should be done on the application level (by calling labels repository) and exact values should be passed to this method.